src/coprocessor/codec/mysql/json/mod.rs

```diff
@@ -95,6 +95,14 @@ pub fn json_object(kvs: Vec<Datum>) -> Result<Json> {
     Ok(Json::Object(map))
 }
 
+impl crate::coprocessor::codec::data_type::AsMySQLBool for Json {
+    #[inline]
+    fn as_mysql_bool(&self) -> bool {
+        // FIXME: Is it correct?
+        false
```

Comment:
According to current code looks like for Json types its bool value will be always false. Is this what we expect? @xuhuaiyu PTAL, thanks!

Reply:
I think we should return an error here.

Reply:
Added an issue to TiDB, looks like this needs to be fixed in TiDB as well: https://github.com/pingcap/tidb/issues/9593